Traceback (most recent call last):
  File "/home/runner/work/retirement/retirement/.venv/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/work/retirement/retirement/.venv/lib/python3.11/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/retirement/retirement/.venv/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/runner/work/retirement/retirement/.venv/lib/python3.11/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/home/runner/work/retirement/retirement/.venv/lib/python3.11/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/work/retirement/retirement/.venv/lib/python3.11/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# two assets, A and B, constant price for A=100 and B=200
prices=pd.read_csv("data/prices.csv", header=0, index_col=0, parse_dates=True)
prices.loc["2022-01-03", "B"] = np.NaN
prices.loc["2022-01-04", "B"] = np.NaN
prices
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
Cell [0;32mIn[2], line 3[0m
[1;32m      1[0m [38;5;66;03m# two assets, A and B, constant price for A=100 and B=200[39;00m
[1;32m      2[0m prices[38;5;241m=[39mpd[38;5;241m.[39mread_csv([38;5;124m"[39m[38;5;124mdata/prices.csv[39m[38;5;124m"[39m, header[38;5;241m=[39m[38;5;241m0[39m, index_col[38;5;241m=[39m[38;5;241m0[39m, parse_dates[38;5;241m=[39m[38;5;28;01mTrue[39;00m)
[0;32m----> 3[0m prices[38;5;241m.[39mloc[[38;5;124m"[39m[38;5;124m2022-01-03[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mB[39m[38;5;124m"[39m] [38;5;241m=[39m [43mnp[49m[38;5;241;43m.[39;49m[43mNaN[49m
[1;32m      4[0m prices[38;5;241m.[39mloc[[38;5;124m"[39m[38;5;124m2022-01-04[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mB[39m[38;5;124m"[39m] [38;5;241m=[39m np[38;5;241m.[39mNaN
[1;32m      5[0m prices

File [0;32m~/work/retirement/retirement/.venv/lib/python3.11/site-packages/numpy/__init__.py:397[0m, in [0;36m__getattr__[0;34m(attr)[0m
[1;32m    394[0m     [38;5;28;01mraise[39;00m [38;5;167;01mAttributeError[39;00m(__former_attrs__[attr])
[1;32m    396[0m [38;5;28;01mif[39;00m attr [38;5;129;01min[39;00m __expired_attributes__:
[0;32m--> 397[0m     [38;5;28;01mraise[39;00m [38;5;167;01mAttributeError[39;00m(
[1;32m    398[0m         [38;5;124mf[39m[38;5;124m"[39m[38;5;124m`np.[39m[38;5;132;01m{[39;00mattr[38;5;132;01m}[39;00m[38;5;124m` was removed in the NumPy 2.0 release. [39m[38;5;124m"[39m
[1;32m    399[0m         [38;5;124mf[39m[38;5;124m"[39m[38;5;132;01m{[39;00m__expired_attributes__[attr][38;5;132;01m}[39;00m[38;5;124m"[39m
[1;32m    400[0m     )
[1;32m    402[0m [38;5;28;01mif[39;00m attr [38;5;241m==[39m [38;5;124m"[39m[38;5;124mchararray[39m[38;5;124m"[39m:
[1;32m    403[0m     warnings[38;5;241m.[39mwarn(
[1;32m    404[0m         [38;5;124m"[39m[38;5;124m`np.chararray` is deprecated and will be removed from [39m[38;5;124m"[39m
[1;32m    405[0m         [38;5;124m"[39m[38;5;124mthe main namespace in the future. Use an array with a string [39m[38;5;124m"[39m
[1;32m    406[0m         [38;5;124m"[39m[38;5;124mor bytes dtype instead.[39m[38;5;124m"[39m, [38;5;167;01mDeprecationWarning[39;00m, stacklevel[38;5;241m=[39m[38;5;241m2[39m)

[0;31mAttributeError[0m: `np.NaN` was removed in the NumPy 2.0 release. Use `np.nan` instead.

